winterfell/services/claimService.js

```diff
@@ -2,18 +2,88 @@ var _ = require('lodash');
 var http = require('http-status-codes');
 var httpErrors = require('./../utils/httpErrors');
 var Claim = require('./../models/claim');
+var Form = require('./../models/form');
+var Q = require('q');
+var bulk = require('bulk-require');
+var formlyFields = bulk(__dirname + '/../forms/', ['*']);
+var expressions = require("angular-expressions");
 
 function ClaimService(app) {
   this.app = app;
 }
 
+/**
+ * For a given form and set of responses, calculate how many questions
+ * were answered (which is simply the number of keys in the responses),
+ * and more complicatedly: how many questions on the form were answerable,
+ * which requires evaluating the angular hideExpression attribute for
+ * each field against the current responses.
+ *
+ * @param formName
+ * @param data
+ * @returns {{answerable: number, answered: number}}
+ */
+function calculateProgress(formName, data) {
+  var evaluate, i;
+  var template = formlyFields[formName];
+  var output = {answerable: 0, answered: _.size(data)};
```

Comment:
_.size(data) as answered could be a dangerous. Let's say we have 5 questions A, B, C, D, E and only A and B are important and required.
If I just answer C and D, `answered` would count those two even though we haven't really completed the form.

Maybe you've gone about things in which every question that is `answerable`, is required. Is that the case? I think there could be a case where you could choose to answer a question, but if you skipped it, it's okay so it's technically optional.